compiler/dyno/lib/uast/post-parse-checks.cpp

```diff
@@ -657,6 +688,86 @@ void Visitor::checkExportedName(const NamedDecl* node) {
   (void) node;
 }
 
+bool Visitor::isNamedThisAndNotReceiverOrFunction(const NamedDecl* node) {
+  if (node->name() != USTR("this")) return false;
+  if (node->isFunction()) return false;
+  if (node->isFormal())
+    if (auto decl = searchParentsForDecl(nullptr))
+      if (auto fn = decl->toFunction())
+        if (fn->thisFormal() == node)
+          return false;
+  return true;
+}
+
+bool Visitor::isNameReservedWord(const NamedDecl* node) {
+  auto name = node->name();
+  if (isNamedThisAndNotReceiverOrFunction(node)) return true;
+  if (name == "none") return true;
+  if (name == "false") return true;
+  if (name == "true") return true;
+  return false;
+}
+
+// TODO: Maybe migrate this to a common place?
+static bool isNameReservedType(UniqueString name) {
+  if (name == USTR("bool")      ||
+      name == USTR("int")       ||
+      name == USTR("uint")      ||
+      name == USTR("real")      ||
+      name == USTR("imag")      ||
+      name == USTR("complex")   ||
+      name == USTR("bytes")     ||
+      name == USTR("string")    ||
+      name == USTR("sync")      ||
+      name == USTR("single")    ||
+      name == USTR("owned")     ||
+      name == USTR("shared")    ||
+      name == USTR("borrowed")  ||
+      name == USTR("unmanaged") ||
+      name == USTR("domain")    ||
+      name == USTR("index")     ||
+      name == USTR("locale")    ||
+      name == USTR("nothing")   ||
+      name == USTR("void"))
+    return true;
+  return false;
+}
+
+// TODO: May have to restrict errors to fire only for functions/formals.
+void Visitor::checkReservedSymbolName(const NamedDecl* node) {
+  auto name = node->name();
+
+  // TODO: Do we really want this sort of exception?
+  if (auto mod = node->toModule())
+    if (mod->kind() == Module::IMPLICIT)
```

Comment:
So is this allowing for modules called, like, `true.chpl`? What are the reasons this is included?

Reply:
In fact, after discussion I think right now we should be skipping all modules. See: https://github.com/chapel-lang/chapel/issues/19197

Feel free to cast your vote there, personally I voted to ban any module/file name that conflicts with a reserved word.